### PR TITLE
Add loom pinning tests

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -169,8 +169,7 @@ subprojects {
 
     if (platform == "loom") {
       jvmArgs(
-        "-Djdk.tracePinnedThread=full",
-        "--enable-preview"
+        "-Djdk.tracePinnedThreads=short",
```
Comment (yschimke) [inline build.gradle.kts:172]: not: the s was missing.

**Comment by yschimke** [comment]
```
    okhttp3.internal.http2.Http2Writer.flush(Http2Writer.kt:125) <== monitors:1
    okhttp3.internal.http2.Http2Connection.newStream(Http2Connection.kt:271) <== monitors:2
Thread[#62,ForkJoinPool-1-worker-3,5,CarrierThreads]
    okhttp3.internal.http2.Http2Stream$FramingSource.read(Http2Stream.kt:395) <== monitors:1
```

**Comment by yschimke** [comment]
@swankjesse Going to land even though it has a failing test for loom. The break is real, and I'd like to fix.

requesting post review, tests and CI only.

File: container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+@Testcontainers
+@Isolated
+class BasicLoomTest {
+  @JvmField
+  @RegisterExtension
+  val platform = PlatformRule()
+
+  @Container
+  val mockServer: MockServerContainer = MockServerContainer(MOCKSERVER_IMAGE)
```
Comment (swankjesse) [inline container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt:38]: Why isn’t this using MockWebServer?!

File: container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    client =
+      OkHttpClient.Builder()
+        .trustMockServer()
+        .dispatcher(Dispatcher(newVirtualThreadPerTaskExecutor()))
+        .build()
+
+    executor = newVirtualThreadPerTaskExecutor()
+
+    System.setOut(PrintStream(capturedOut))
```
Comment (swankjesse) [inline container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt:59]: What’s this for?

**Review by swankjesse — COMMENTED** [review]
LGTM

File: container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt (hunk 2 above)
Comment (yschimke) [reply container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt:38]: I'll put a comment. I only wanted to fail on the client side. Not to test our mockwebserver is loom compliant also.

File: container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt (hunk 3 above)
Comment (yschimke) [reply container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt:59]: I don't have a programmatic way to detect loom blocking. Just capture this warning.

File: container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt (hunk 2 above)
Comment (swankjesse) [reply container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt:38]: Makes sense. I’d like to also fix MockWebServer to be compliant!

File: container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt (hunk 3 above)
Comment (swankjesse) [reply container-tests/src/test/java/okhttp3/containers/BasicLoomTest.kt:59]: Gotcha.

**Pull request merged**